front/app/api/campaign_consents/useCampaignConsents.ts

```diff
@@ -2,27 +2,37 @@ import { useQuery } from '@tanstack/react-query';
 import { CLErrors } from 'typings';
 import fetcher from 'utils/cl-react-query/fetcher';
 import campaignConsentKeys from './keys';
-import { CampaignConsentKeys, ICampaignConsents } from './types';
+import {
+  CampaignConsentKeys,
+  ICampaignConsents,
+  IConsentsRequestData,
+} from './types';
 
-const fetchCampaignConsents = (unsubscriptionToken) =>
-  fetcher<ICampaignConsents>({
+const fetchCampaignConsents = (consentsRequestData: IConsentsRequestData) => {
```

Comment:
```suggestion
const fetchCampaignConsents = (consentsRequestData?: IConsentsRequestData) => {
```
Maybe better so we don't need to make calls like `useCampaignConsents({})`?